packages/slice-machine/src/modules/pushSaga.ts

```diff
@@ -0,0 +1,149 @@
+    if (axios.isAxiosError(e) && e.response?.status) {
+      return e.response.status;
+    }
+    return slice.model.id;
+  }
+}
+
+function* pushCustomTypeSaga(customType: CustomTypeSM) {
+  try {
+    yield call(pushCustomType, customType.id);
+    yield put(pushCustomTypeCreator.success({ customTypeId: customType.id }));
+    return undefined;
+  } catch (e) {
+    if (axios.isAxiosError(e) && e.response?.status) {
+      return e.response.status;
+    }
+    return customType.id;
+  }
+}
+
+function isTruthyString(str: unknown): str is string {
+  return typeof str === typeof "" && Boolean(str);
+}
+
+function isAuthErrorCode(n: unknown): n is number {
+  return typeof n === typeof 0 && (n === 401 || n === 403);
+}
+
+export function* changesPushSaga({
+  payload,
+}: ReturnType<typeof changesPushCreator>) {
+  const { unSyncedSlices, unSyncedCustomTypes } = payload;
+
+  // eslint-disable-next-line @typescript-eslint/no-unsafe-assignment
+  const slicesPushResults: Array<string | undefined | number> = yield all(
+    unSyncedSlices.map((slice) => call(pushSliceSaga, slice))
+  );
+
+  const sliceAuthFailures = slicesPushResults.filter(isAuthErrorCode).length;
+  const slicesFailures = slicesPushResults.filter(isTruthyString).length;
+
+  if (sliceAuthFailures > 0) {
+    yield put(modalOpenCreator({ modalKey: ModalKeysEnum.LOGIN }));
+  } else if (slicesFailures > 0) {
+    const errorMsg = `Failed to upload ${slicesFailures} slice${
+      slicesFailures > 1 ? "s" : ""
+    }`;
+    yield put(
+      openToasterCreator({
+        message: errorMsg,
+        type: ToasterType.ERROR,
+      })
+    );
+  }
+
+  // eslint-disable-next-line @typescript-eslint/no-unsafe-assignment
+  const customTypePushResults: Array<string | undefined | number> = yield all(
+    unSyncedCustomTypes.map((ct) => call(pushCustomTypeSaga, ct))
+  );
+
+  const customTypeAuthFailures =
+    customTypePushResults.filter(isAuthErrorCode).length;
+  const customTypeFailures =
+    customTypePushResults.filter(isTruthyString).length;
+
+  if (customTypeAuthFailures > 0) {
+    yield put(modalOpenCreator({ modalKey: ModalKeysEnum.LOGIN }));
+  } else if (customTypeFailures > 0) {
+    const errorMsg = `Failed to upload ${customTypeFailures} custom type${
+      customTypeFailures > 1 ? "s" : ""
+    }`;
+    yield put(
+      openToasterCreator({
+        message: errorMsg,
+        type: ToasterType.ERROR,
+      })
+    );
+  }
+
+  if (
+    sliceAuthFailures +
+      slicesFailures +
+      customTypeFailures +
+      customTypeAuthFailures ===
+    0
+  ) {
+    yield put(
+      openToasterCreator({
+        message: "All slices and custom types have been pushed",
+        type: ToasterType.SUCCESS,
+      })
+    );
+  }
+
+  const { data: serverState } = (yield call(getState)) as SagaReturnType<
+    typeof getState
+  >;
+  const { customTypes, ...rest } = serverState;
+  yield put(refreshStateCreator({ ...rest, localCustomTypes: customTypes }));
```

Comment:
Here instead of doing fetching the state manually and then refreshing the state manually, we can just use SWR to force the refresh of the state.

https://stackoverflow.com/questions/61653622/next-react-swr-refresh-from-button-click-in-child-component-can-i-use-a-call

Reply:
I'd create like a "forceRefreshState" in the `useServerState` file and export it to be used wherever we need.

Reply:
I'll have to look into that.